### ISSUE TITLE
When I swipe to open item, it auto closed in Android version KitKat and above.

### ISSUE THREAD

**Issue by surajshrestha** [daimajia/AndroidSwipeLayout#412]
In my project I have build this version of library  "com.daimajia.swipelayout:library:1.2.0@aar". Every time i swipe left to open hidden buttons it also closed ?? Does anyone came across this issue and solve it ??

**Comment by zung** [comment]
i have this issues too.

**Comment by daimajia** [comment]
@surajshrestha Make sure that you are using the internal adapter instead of your own. Check this sample https://github.com/daimajia/AndroidSwipeLayout/wiki/SwipeAdapter#example

**Comment by daimajia** [comment]
@zung

**Comment by masuikvova** [comment]
the same problem

**Comment by bhavna935** [comment]
even i am facing the same issue..

**Comment by bhavna935** [comment]
@surajshrestha,
Did you find any solution for this?

**Comment by pulsar314** [comment]
I am having the same problem on Android Lollipop. After item had been swiped to open, onLayout is called, that closes item. Maybe you need to save state between onLayout calls?

**Comment by AllenWen** [comment]
how to use SwipeAdapter for ExpandableListView?

**Comment by RammiyaBaskar** [comment]
@AllenWen  use it as interface like this
  `public class RecyclerAdaptert extends ExpandableRecyclerAdapter<> implements SwipeAdapterInterface,SwipeItemMangerInterface {}
`

**Comment by TruthKeeper** [comment]
exactly,in my SAMSUNG 7(Andorid 7.0) it's ok,but in HTC(Android 5.0) exist some problem,it's hard to slide the menu,and it will be close automatically,by the way ,extends com.daimajia.swipe.adapters.RecyclerSwipeAdapter can‘t change result

**Comment by reinhardt1053** [comment]
I have the same issue.

**Comment by asadwaheed1** [comment]
I also have the same issue.

**Comment by Pahanuch** [comment]
How to fix it ?

**Comment by cilvet** [comment]
Same issue here.

**Comment by abhishekBansal** [comment]
Facing this issue only on Motorola Moto E3 Power Device with Android Marshmallow 6.0. Tested on many other configurations. It works fine.

**Comment by aik117** [comment]
i have tried [this](https://github.com/daimajia/AndroidSwipeLayout/issues/357) solution and many others but whenever i call textview.settext in onUpdate method the swipe closes itself automatically. I have even added the library as a module in my project and commented the method safeBottomViews() in updateBottomViews() in SwipeLayout.java but nothing seems to work. This library is very good no doubt but this issue is getting on my nerves now. Please if anyone got a solution then let me know. Thanks.

**Comment by juancamachodev** [comment]
my problem was in the adapter:
Adapter extends RecyclerView.Adapter <Adapter.MyViewHolder>
the solution was this
Adapter extends RecyclerSwipeAdapter <Adapter.MyViewHolder>

     @Override
     public int getSwipeLayoutResourceId (int position) {
         return R.id.swipe;
     }

and in the onBindViewHolder add method

mItemManger.bindView (holder.itemView, position);

definitely this is not google translator :v

**Comment by Drake1804** [comment]
Find another one solution that works for me.

```
 swipeLayout.addSwipeListener(new SimpleSwipeListener() {
        @Override
        public void onStartOpen(SwipeLayout layout) {
            isOpenSwipeLayout = true;
        }

        @Override
        public void onStartClose(SwipeLayout layout) {
            isOpenSwipeLayout = false;
        }
    });
    swipeGlobalLayoutListener = new ViewTreeObserver.OnGlobalLayoutListener() {
        @Override
        public void onGlobalLayout() {
            if (isOpenSwipeLayout) {
                // Opens the layout without animation
                swipeLayout.open(false);
            }
        }
    };
swipeLayout.getViewTreeObserver().addOnGlobalLayoutListener(swipeGlobalLayoutListener);
```

isOpenSwipeLayout - global variable
